File: 0071A.Way.Too.Long.Words/solution.py
 def solution():
    case_count = int(input())

    for _ in range(case_count):
        word = input()
        if len(word) <= 10:
            print(word)
        else:
            print(f"{word[0]}{len(word) - 2}{word[-1]}")


 if __name__ == "__main__":
      solution()

File: 0071A.Way.Too.Long.Words/solution.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var reader *bufio.Reader = bufio.NewReader(os.Stdin)
var writer *bufio.Writer = bufio.NewWriter(os.Stdout)

func Printf(format string, x ...interface{}) {
	fmt.Fprintf(writer, format, x...)
}

func Scanf(format string, x ...interface{}) {
	fmt.Fscanf(reader, format, x...)
}

func Solution() {
	var caseCount int
	Scanf("%d\n", &caseCount)

	var word string
	for caseIndex := 0; caseIndex < caseCount; caseIndex++ {
		Scanf("%s\n", &word)
		if len(word) <= 10 {
			Printf("%s\n", word)
		} else {
			Printf("%c%d%c\n", word[0], len(word)-2, word[len(word)-1])
		}
	}
}

func main() {
	Solution()
	writer.Flush()
}
